AsyncDisplayKit/Private/ASTranslationRange.mm

```diff
@@ -0,0 +1,158 @@
+//
+//  ASTranslationRange.mm
+//  AsyncDisplayKit
+//
+//  Created by Levi McCallum on 4/4/16.
+//  Copyright © 2016 Facebook. All rights reserved.
+//
+
+#import "ASTranslationRange.h"
+
+#import "ASAssert.h"
+#import "ASThread.h"
+#import <vector>
+#import <algorithm>
+
+@interface ASTranslationRange () {
+  NSUInteger _length;
+  std::vector<NSRange> _ranges;
+  ASDN::RecursiveMutex _propertyLock;
```

Comment:
It could be a good idea to make sure this object is internally locked, but if all of the usages of it are already locked by the owning object, then this is not needed.